stan/math/rev/core/arena_matrix.hpp

```diff
@@ -280,7 +330,7 @@ class arena_matrix<MatrixType, require_eigen_sparse_base_t<MatrixType>>
   template <typename Expr,
             require_not_same_t<Expr, arena_matrix<MatrixType>>* = nullptr>
   arena_matrix& operator=(Expr&& expr) {
-    *this = arena_matrix(std::forward<Expr>(expr));
+    new (this) arena_matrix(std::forward<Expr>(expr));
```

Comment:
Mind walking me through this? I think I know what it's doing, but I want to double check. 

(Nothing too long; even just a couple of sentences.)